### PR TITLE
Added "setup environment and tooling" section to contribution guidelines

### LINKED ISSUE #1422: [Documentation] Write "Setup environment and tooling for contribution"-section for contribution guidelines
**This is part of a larger issue - please read: https://github.com/kirbydesign/designsystem/issues/1393#issuecomment-802766408 before continuing!**

**Short description**

A "Setup environment and tooling for contribution" section should be added to the contribution guidelines.  An outline already exists on the branch [`documentation/contribution-guidelines`](https://github.com/kirbydesign/designsystem/tree/documentation/contribution-guidelines) which should be expanded upon.

### PR DESCRIPTION
**Pull request by MadsBuchmann**
This PR closes #1422 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #1422 

## What is the new behavior?
A section detailing how to clone the repository, which node version should be used, and which tools are used has been added. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -99,58 +99,69 @@ If you got an idea for a component that you think should be part of Kirby you sh
...
+### Node 
 <!-- Which version should people use, and perhaps a recommendation on which tool people can use to manage versions. -->
 [Node](https://nodejs.org/en/) is the runtime used for development. 
+
 We recommend using Node version `INSERT HIGHEST VERSION WE HAVE GOTTEN RUNNING WITHOUT ISSUES`. This is because we have experienced errors with other version due to **What was the problem exactly again?**.  
 
 If you have need other versions of Node installed then a version manager such as [NVM](https://github.com/nvm-sh/nvm) can be used. 
 
-#### Tslint 
+### Tslint
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:153]: Capital `S`

TSLint instead of Tslint.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -99,58 +99,69 @@ If you got an idea for a component that you think should be part of Kirby you sh
...
 
 To solve this: 
 1. open a typescript file in VScode. 
-2. open the command pallette (<kbd>ctrl/cmd</kbd> + <kbd>shift</kbd> + <kbd>p</kbd>) and enter: `typescript: select typescript version`.
-3. select: `use workspace version 3.6.5`.
-4. open the command pallette (<kbd>ctrl/cmd</kbd> + <kbd>shift</kbd> + <kbd>p</kbd>) and enter: `TSLint: Manage workspace library execution`.
-5. select: `always enable workspace library execution`.
+2. open the command pallette (<kbd>ctrl/cmd</kbd> + <kbd>shift</kbd> + <kbd>p</kbd>) and enter: <br />
+`typescript: select typescript version`.
+4. select: `use workspace version 3.6.5`.
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:178]: Once the Angular (and Typescript) upgrade issue is merged I don't think this version number is correct any more.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -5,189 +5,297 @@ If you at any point have any questions please read: [I have a question related t
...
+* [Enabling "fix on save" in VSCode](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin#auto-fix-and-auto-fix-on-save).
+
+#### VSCode "workspace library execution"-issue
+
+An issue might occur where TSLint will randomly disable for the current workspace and has to be enabled again using the command `TSLint: Manage workspace library execution` . 
+
+To solve this: 
+
+1. open a typescript file in VScode. 
+2. open the command pallette (<kbd>ctrl/cmd</kbd> + <kbd>shift</kbd> + <kbd>p</kbd>) and enter: <br />
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:174]: 👍  for using `<kbd>`

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -5,189 +5,297 @@ If you at any point have any questions please read: [I have a question related t
...
+
+`typescript: select typescript version` .
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:176]: This look a bit odd when displayed. I'm not sure how it's handled in Markdown but I think this line (and similar lines) should be indented, since they're inside a list.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -5,189 +5,297 @@ If you at any point have any questions please read: [I have a question related t
...
+* [TSLint extension for VSCode](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin).
+* [Enabling "fix on save" in VSCode](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin#auto-fix-and-auto-fix-on-save).
+
+#### VSCode "workspace library execution"-issue
+
+An issue might occur where TSLint will randomly disable for the current workspace and has to be enabled again using the command `TSLint: Manage workspace library execution` . 
+
+To solve this: 
+
+1. open a typescript file in VScode.
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:173]: Capitalize the first letter (all items in this list)

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -5,189 +5,297 @@ If you at any point have any questions please read: [I have a question related t
...
+`TSLint: Manage workspace library execution` .
+
+6. select: `always enable workspace library execution`.
+
+This will cause changes to be made in the file `.vscode/settings.json` ; do not commit these.
 
 ### Prettier
-- It is a good idea to use format-on-save
+
+Prettier is used for consistent code formatting and avoiding unnecessary discussions about this during reviews.
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:189]: Consider adding a sentence about reducing "diff clutter".

When everything is formatted the same way using Prettier there should be no changes that are "formatting only".

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -5,189 +5,297 @@ If you at any point have any questions please read: [I have a question related t
...
+
+It can be executed using `npx prettier [path/to/file]` but as with TSLint, we recommend installing it as a plugin for your editor. You should also enable "format on save".
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:191]: I think we have npm script for prettier, too: `npm run prettier` and `npm run prettier:fix`. Consider to mention those.

**Review by jkaltoft — CHANGES_REQUESTED** [review]
Good stuff!

I've requested a few minor changes.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:235]: <kbd>T</kbd><kbd>h</kbd><kbd>a</kbd><kbd>n</kbd><kbd>k</kbd><kbd>space</kbd><kbd>y</kbd><kbd>o</kbd><kbd>u</kbd>

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:239]: I will add a repeating todo to my todo-system, to check up on the angular upgrade issue, and update this if necessary :D

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Should be fixed by new commits

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Should be fixed by new commits

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Should be fixed by new commits

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Couldn't really get markdown to play nice - so I've inlined them. I think it looks _good enough_:
Screenshot 2021-05-04 at 12 54 17

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:246]: How's this? (something 

<hr />

As with TSLint, we  strongly recommend installing Prettier as a plugin for your editor. You should also enable "format on save".

It can however be used without an editor plugin; run `npm run prettier` to check for formatting errors, fix them by running `npm run prettier:fix`. 

If there are changes only concerning formatting, in files you did not create - do not commit these. There is most likely something wrong with your Prettier settings. 

Make sure that Prettier uses the config file that is shipped with the repository ([`.prettierrc.json`](https://github.com/kirbydesign/designsystem/blob/master/.prettierrc.json)) and not a global configuration located on your machine.

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (jkaltoft) [reply .github/CONTRIBUTING.md]: Alright. When I reviewed it it didn't look as nice... 🤷‍♂️. But it looks fine in the screenshot above 👍

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:246]: 👍

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -172,22 +172,93 @@ If you have found a bug in Kirby you should follow these steps:
...
+
+As with TSLint, we  strongly recommend installing Prettier as a plugin for your editor. You should also enable "format on save".
+It can however be used without an editor plugin; run `npm run prettier` to check for formatting errors, fix them by running `npm run prettier:fix`. 
+
+If there are changes only concerning formatting, in files you did not create - do not commit these. There is most likely something wrong with your Prettier settings. 
+Make sure that Prettier uses the config file that is shipped with the repository ([`.prettierrc.json`](https://github.com/kirbydesign/designsystem/blob/master/.prettierrc.json)) and not a global configuration located on your machine. 
+
+To check if the config file from the repository is used, try to alter the configuration in `.prettierrc.json` and format a file using Prettier - for example by changing `"singleQuote": true,` to `"singleQuote": false,`. 
+If all quotations marks are changed to `"` instead of `'` when you format, it means the repository config file is used by Prettier. Make sure revert the changes to `.prettierrc.json`.
```
Comment (jkaltoft) [inline .github/CONTRIBUTING.md:255]: > Make sure [**to**|**you**] revert the changes to `.prettierrc.json`.

Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Changed to:
> To check if the config file from the repository is used, try to alter the configuration in `.prettierrc.json` and format a file using Prettier - for example by changing `"singleQuote": true,` to `"singleQuote": false,`. 
> If all quotation marks are changed to `"` instead of `'` when you format, it means the repository config file is used by Prettier. Remember to revert the changes made to `.prettierrc.json`.

**Review by jkaltoft — APPROVED** [review]
